### ISSUE TITLE
Disparity and flow results not so good on custom data

### ISSUE THREAD

**Issue by poornimajd** [visinf/self-mono-sf#4]
I have trained the model on my custom data with 5000 stereo pairs and 20 epochs.I got the following results
Input image
![000000_10](https://user-images.githubusercontent.com/31023599/85939814-02871080-b936-11ea-9e1b-cb0374a988dc.png)
Output disparity: gray and colored respectively as shown:
![000000_10b](https://user-images.githubusercontent.com/31023599/85939826-17fc3a80-b936-11ea-89fb-e0bf34529a51.png)
![000000_disp](https://user-images.githubusercontent.com/31023599/85939827-19c5fe00-b936-11ea-9607-f704c340eb95.jpg)
Flow:
![000000_flow](https://user-images.githubusercontent.com/31023599/85939833-264a5680-b936-11ea-97f4-d8ed507cfb2a.png)
What can be the reason for this?
Could this also happen if the correct focal length is higher by 20 units as compared to that given for training?
Any suggestion is appreciated!
Thanks!

**Comment by hurjunhwa** [comment]
Hi, 
I am not sure it's due to the different focal length.
By looking at the visualized optical flow outputs, it seems that the network is not converged yet. 
We trained the network for 400k iteration steps with ~26k stereo pairs. 
(which is 400k / (26k pairs / 4 mini-batch size ) = approximately 62 epochs)
Could you maybe train the networks upto 400k iteration steps with your custom data?

**Comment by poornimajd** [comment]
Thanks for the reply.I will train it for some more epochs and check the results.After training it for 20 epochs these are the loss graphs I obtained.
![selfmono](https://user-images.githubusercontent.com/31023599/86339905-c6bfb400-bc71-11ea-8d2a-d628dce6c35d.png)
The below graph is on the validation images if I am not wrong and it has lower loss compared to that of the train loss,this might be due to the validation images being easier right?

**Comment by hurjunhwa** [comment]
Yes I think so too.
You may also do a overfitting test: using only few image samples, trying to overfit on them, and looking at the visualization of the training samples. 
If it doesn't overfit, I guess there can be some issues in the custom dataloader or input data..
